### PR TITLE
PlaneDetector common interface.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: proxy_mobile/magic-script/MagicScript.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import { AppRegistry } from 'react-native';
-import { ReactNativeMagicScript, PlatformInformation } from 'magic-script-components-react-native';
+import { ReactNativeMagicScript, PlatformInformation, NativePlaneDetector } from 'magic-script-components-react-native';
```
Comment (pleszkiewicz) [inline proxy_mobile/magic-script/MagicScript.js]: How about using prefix `ReactNative`?
`ReactNativeMagicScript`, `ReactNativePlatformInformation`, `ReactNativePlaneDetection`?

File: src_app/catalog_app.js [hunk 2]
```diff
@@ -89,9 +90,11 @@ class CatalogApp extends React.Component {
       { name: 'Clock', component: <SceneClock localPosition={[0, 0, 0]} /> },
       { name: 'Line', component: <SceneLine localPosition={[0, 0, 0]} /> },
       // { name: 'File Picker', component: <SceneFilePicker localPosition={[0, 0, -0.5]} /> },
+      { name: 'Plane Detector', component: <ScenePlaneDetector localPosition={[0, 0, -0.5]} /> },
```
Comment (pleszkiewicz) [inline src_app/catalog_app.js]: Does it work on Android and Lumin?

File: src_app/test_scenes/index.js [hunk 3]
```diff
@@ -37,4 +37,5 @@ export * from './sceneTransformations';
 export * from './sceneVideo';
 export * from './sceneVideoGrid';
 export * from './sceneWebView';
-export * from './sceneFilePicker'
\ No newline at end of file
+export * from './sceneFilePicker';
+export * from './scenePlaneDetector';
```
Comment (pleszkiewicz) [inline src_app/test_scenes/index.js]: Can we put all the exports in alphabetical order?

File: src_app/test_scenes/scenePlaneDetector.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const { points } = this.state;
+    for (var key in points) {
+      lines.push(this.renderLine(points[key]));
+    }
+    return lines;
+  }
+
+  render() {
+    return (
+      <view localPosition={this.props.localPosition}>{this.renderLines()}</view>
```
Comment (pleszkiewicz) [inline src_app/test_scenes/scenePlaneDetector.js]: Is it a good idea to move origin of this scene? All planes will be shifted, right?

File: src_app/test_scenes/scenePlaneDetector.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      } else {
+        console.log("getAllPlanes", result);
+      }
+    });
+  }
+
+  onPlaneDetected = event => {
+    console.log("onPlaneDetected", event);
+    // const { points } = this.state;
+    // points[event.id] = event.vertices;
```
Comment (pleszkiewicz) [inline src_app/test_scenes/scenePlaneDetector.js]: Why does the event handlers are commented out? That means that this test scene does not work, does it?

**Review by pleszkiewicz — COMMENTED** [review]
Code: the test scene is not finished, there is the missing part responsible for points managment.

**Comment by panwrona** [comment]
Hey, if I merge the new catalog files structure first, you will have to change below files:
> Code from `proxy_mobile/magic-script/MagicScript.js` to `common/magic-script/MagicScript.js`
> Code from `src/main.js` to `lumin/src/main.js`
> Code from `src_app` to `src`

File: proxy_mobile/magic-script/MagicScript.js (hunk 1 above)
Comment (SPaluchiewiczSofomo) [reply proxy_mobile/magic-script/MagicScript.js]: I'll prepare this change as separate PR.

File: src_app/test_scenes/scenePlaneDetector.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      index++;
+    }
+    return lines;
+  }
+
+  render() {
+    const { planes } = this.state;
+    console.log("Number of planes", Object.keys(planes).length);
+    return (
+      <view localScale={[2,2,2]}>{this.renderLines()}</view>
```
Comment (pleszkiewicz) [inline src_app/test_scenes/scenePlaneDetector.js]: It may be not obvious why we use scale [2,2,2] here.

File: src_app/test_scenes/index.js [hunk 7]
```diff
@@ -8,16 +8,20 @@ export * from './sceneColorPicker';
 export * from './sceneDatePicker';
 export * from './sceneDialog';
 export * from './sceneDropdownList';
+export * from './sceneFilePicker';
 export * from './sceneGridLayout';
 export * from './sceneImage';
-export * from './sceneImageSlider/scene';
 export * from './sceneImageRemote';
+export * from './sceneImageSlider/scene';
```
Comment (pleszkiewicz) [inline src_app/test_scenes/index.js]: We don't need to provide `/scene`:

```export * from './sceneImageSlider';```

**Review by pleszkiewicz — APPROVED** [review]
Code: 👍

File: src_app/test_scenes/index.js (hunk 7 above)
Comment (SPaluchiewiczSofomo) [reply src_app/test_scenes/index.js]: Looks like post rebase leftover - cleaned-up.

**Pull request merged**